### ISSUE TITLE
在使用 `model.val(..., save_txt=True, ...)` 进行分割评估时，导出的 txt 文件中的多边形坐标存在明显的"整体往图像中心内缩"现象，特别是贴近图像边缘的目标。而使用 `model.predict(..., save_txt=True, ...)` 导出的 txt 文件与 ground truth 基本一致。

### ISSUE THREAD

**Issue by mingren8888** [ultralytics/ultralytics#23584]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

在使用 `model.val(..., save_txt=True, ...)` 进行分割评估时，导出的 txt 文件中的多边形坐标存在明显的"整体往图像中心内缩"现象，特别是贴近图像边缘的目标。而使用 `model.predict(..., save_txt=True, ...)` 导出的 txt 文件与 ground truth 基本一致。

### Environment

- **操作系统**：Windows 10 (Build 26200)
- **Python 版本**：3.11.9
- Ultralytics**：8.3.86 (YOLOv11)

### Minimal Reproducible Example

from ultralytics import YOLO

# 加载模型
model = YOLO("yolo11n-seg.pt")

# 方法1：使用 val 导出 txt（有问题）
model.val(
    data="dataset.yaml",
    imgsz=640,
    save_txt=True,
    save_conf=True,
    project="runs/segment",
    name="val_test",
)

# 方法2：使用 predict 导出 txt（正确）
model.predict(
    source="path/to/val/images",
    imgsz=640,
    save_txt=True,
    save_conf=True,
    retina_masks=False,
    project="runs/segment",
    name="predict_test",
)
### 1.3 预期行为

`val(save_txt=True)` 和 `predict(save_txt=True)` 导出的 txt 文件应该一致，多边形坐标应该与 ground truth 对齐（允许 1-2 像素的正常磨边）。

### 1.4 实际行为

`val(save_txt=True)` 导出的 txt 文件中，多边形坐标明显内缩，特别是贴近图像边缘的目标。例如：
- 顶边从 0.0 缩到 ~0.036+
- 与 `predict` 路径相比，`predict` 只缩到 ~0.003 左右

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mingren8888, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the detailed report and code snippet—this does look like a potential 🐛 discrepancy between `val(save_txt=True)` and `predict(save_txt=True)` export behavior for segmentation polygons (especially near image borders). This is an automated response 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. You’ve already shared a great start—could you please add the missing pieces needed to reproduce end-to-end?
-Your `dataset.yaml` (or a sanitized version) and the exact `data=` path used
-1-3 sample validation images that show the border effect clearly
-The corresponding ground-truth label files for those images
-The exact output `.txt` from both `runs/segment/val_test/` and `runs/segment/predict_test/` for the same images
-Your full console logs for both commands (including any preprocessing/letterbox notes)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
They run at different imgsz if you don't use `rect=False`

https://github.com/ultralytics/ultralytics/issues/2783#issuecomment-2413530537

**Comment by Y-T-G** [comment]
Also this Ultralytics version is too old
